test/InProcessTestFramework/MockDurableTaskClient.cs

```diff
@@ -0,0 +1,122 @@
+/// without requiring a full backend service.
+/// </summary>
+public sealed class MockDurableTaskClient : DurableTaskClient
+{
+    readonly InProcessTestOrchestrationService orchestrationService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MockDurableTaskClient"/> class.
+    /// </summary>
+    /// <param name="orchestrationService">The in-process orchestration service.</param>
+    public MockDurableTaskClient(InProcessTestOrchestrationService orchestrationService)
+        : base("MockDurableTaskClient")
+    {
+        this.orchestrationService = orchestrationService ?? throw new ArgumentNullException(nameof(orchestrationService));
+    }
+
+    /// <inheritdoc/>
+    public override async Task<string> ScheduleNewOrchestrationInstanceAsync(
+        TaskName orchestratorName, 
+        object? input = null, 
+        StartOrchestrationOptions? options = null, 
+        CancellationToken cancellation = default)
+    {
+        var instanceId = options?.InstanceId ?? Guid.NewGuid().ToString("N");
+        
+        await this.orchestrationService.ScheduleOrchestrationAsync(
+            orchestratorName, 
+            instanceId, 
+            input,
+            options,
+            cancellation);
+            
+        return instanceId;
+    }
+
+    /// <inheritdoc/>
+    public override Task<OrchestrationMetadata> GetInstanceAsync(
+        string instanceId, 
+        bool getInputsAndOutputs = false, 
+        CancellationToken cancellation = default)
+    {
+        return this.orchestrationService.GetInstanceAsync(instanceId, getInputsAndOutputs, cancellation);
+    }
+
+    /// <inheritdoc/>
+    public override Task<OrchestrationMetadata> WaitForInstanceStartAsync(
+        string instanceId, 
+        bool getInputsAndOutputs = false, 
+        CancellationToken cancellation = default)
+    {
+        return this.orchestrationService.WaitForInstanceStartAsync(instanceId, getInputsAndOutputs, cancellation);
+    }
+
+    /// <inheritdoc/>
+    public override Task<OrchestrationMetadata> WaitForInstanceCompletionAsync(
+        string instanceId, 
+        bool getInputsAndOutputs = false, 
+        CancellationToken cancellation = default)
+    {
+        return this.orchestrationService.WaitForInstanceCompletionAsync(instanceId, getInputsAndOutputs, cancellation);
+    }
+
+    /// <inheritdoc/>
+    public override Task RaiseEventAsync(
+        string instanceId, 
+        string eventName, 
+        object? eventPayload = null, 
+        CancellationToken cancellation = default)
+    {
+        return this.orchestrationService.RaiseEventAsync(instanceId, eventName, eventPayload, cancellation);
+    }
+
+    /// <inheritdoc/>
+    public override Task SuspendInstanceAsync(string instanceId, string? reason = null, CancellationToken cancellation = default)
+    {
+        return this.orchestrationService.SuspendInstanceAsync(instanceId, reason, cancellation);
```

Comment:
I just realized that I shouldn't write suspend/resume like this  as we won't be able to pause the running task. I would just throw not supported exception here for now